tests/performance-tests/src/configuration/spikeProfile.ts

```diff
@@ -0,0 +1,84 @@
+import { Options } from 'k6/options';
+import { mergeObjects, parseIntOptional } from '../utils/utils';
+
+interface Config {
+  // Duration of normal traffic period prior to spike.
+  preSpikeStageDurationMinutes: number;
+  // Duration of spike traffic period.
+  spikeStageDurationMinutes: number;
+  // Duration of a short final stage to allow any long-running requests to complete.
+  postSpikeStageDurationMinutes: number;
+  // Normal traffic levels before and after spike.
+  normalTrafficRequestRatePerSecond: number;
+  // The maximum requests per second to ramp up to during the traffic spike.
+  spikeRequestRatePerSecond: number;
+}
+
+const overrides: Partial<Config> = {
+  preSpikeStageDurationMinutes: parseIntOptional(__ENV.PRE_SPIKE_DURATION_MINS),
+  spikeStageDurationMinutes: parseIntOptional(__ENV.SPIKE_DURATION_MINS),
+  postSpikeStageDurationMinutes: parseIntOptional(
+    __ENV.POST_SPIKE_DURATION_MINS,
+  ),
+  normalTrafficRequestRatePerSecond: parseIntOptional(__ENV.RPS_NORMAL),
+  spikeRequestRatePerSecond: parseIntOptional(__ENV.RPS_SPIKE),
+};
+
+export default function spikeProfile({
+  scenario,
+  ...defaultConfig
+}: Config & { scenario?: string }): Options {
+  const {
+    preSpikeStageDurationMinutes,
+    spikeStageDurationMinutes,
+    postSpikeStageDurationMinutes,
+    normalTrafficRequestRatePerSecond,
+    spikeRequestRatePerSecond,
+  } = mergeObjects(defaultConfig, overrides);
+
+  return {
+    scenarios: {
+      [scenario ?? 'spike']: {
```

Comment:
Could define this default in the function parameter?

```ts
export default function spikeProfile({
  scenario = 'spike',
})
```

Reply:
Hah - I didn have ti that way originally - will change back ta

Reply:
Done ta.  Already had it this way in `rampingRequestRateProfile` but have changed in `steadyRequestRateProfile` too.